AutoRest/Generators/CSharp/Azure.CSharp/AzureCSharpCodeNamer.cs

```diff
@@ -24,15 +24,31 @@ public class AzureCSharpCodeNamer : CSharpCodeNamer
             // Do nothing
         }
 
-        private static string GetNextLinkString(Dictionary<string, object> extensions)
+        private static string GetPagingSetting(ServiceClient serviceClient, Dictionary<string, object> extensions, out string nextLinkName)
         {
+            // default value
+            nextLinkName = null;
             var ext = extensions[AzureCodeGenerator.PageableExtension] as Newtonsoft.Json.Linq.JContainer;
             if (ext == null)
             {
                 return null;
             }
 
-            return (string)ext["path"];
+            nextLinkName = (string)ext["nextLinkName"] ?? "nextLink";
+            string itemName = (string)ext["itemName"] ?? "value";
+            var keypair = new KeyValuePair<string, string>(nextLinkName, itemName);
+            if (!serviceClient.PageClasses.ContainsKey(keypair))
+            {
+                if (serviceClient.PageClasses.Count > 0)
```

Comment:
Actually, the code will generate multiple page classes (named: Page, Page1, Page2, ...) if there are multiple x-page extensions with different "nextLinkName" or "itemName".